### ISSUE TITLE
Fix pending Signup system test

### ISSUE THREAD

**Issue by garettarrowood** [rubyforgood/stocks-in-the-future#57]
# Summary

We have only one system test at the moment. We will need many more. But first we need to get our first one back into a working state.

It originally was passing (of course), then through many contributions over a Ruby4Good event, something went wrong.

## Things to Consider

```
Failures:

  1) Users sign up flow works
     Failure/Error: fill_in "Email", with: "test123@test.com"

     Capybara::ElementNotFound:
       Unable to find field "Email" that is not disabled
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:312:in `block in synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/base.rb:84:in `synchronize'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:301:in `synced_resolve'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/finders.rb:60:in `find'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/node/actions.rb:91:in `fill_in'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/session.rb:773:in `fill_in'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `call'
     # /usr/local/bundle/gems/capybara-3.39.2/lib/capybara/dsl.rb:52:in `fill_in'
     # ./spec/system/users_spec.rb:12:in `block (3 levels) in <main>'

Finished in 3.07 seconds (files took 2.95 seconds to load)
15 examples, 1 failure
```

# Criteria for Completion

- [ ] Test is fixed so that it asserts a working signup flow.

# Bonus

- [ ] Addition of any existing user login and/or password update system tests.

**Comment by napster235** [comment]
Hi @garettarrowood. I looked at this issue and I have a different error for this test that is caused by the fact that the `Sign up` page has only 3 fields(email, password, and password confirmation). However, the User table has 3 more fields(first_name, last_name, and username) that are required.

The errors are shown on the page after clicking the 'Sign up' button.

![Screenshot 2023-09-22 at 12 42 00](https://github.com/rubyforgood/stocks-in-the-future/assets/20145349/d09287d5-91ba-4b57-b25c-d2b507415f0e)

If the validations from the User model are left out, the test passes.
Probably the devise view should be changed in order to accommodate the missing fields from the form.

**Comment by garettarrowood** [comment]
@napster235 - Thanks for the callouts here, and the PR!

We're waiting on the final confirmation of the design of the new login screen. Here's the old site:

<img width="744" alt="Screenshot 2023-09-22 at 9 45 34 AM" src="https://github.com/rubyforgood/stocks-in-the-future/assets/12770108/1ff8e8ab-e048-4da5-a62f-8ed3a9f0c9a1">

I think users may sometimes use their email as their username, which is separate from their name(s), but the two examples I have are that username is different. I'll get this question to the stakeholder to see if we want to change it to email, because that seems like less for the users to remember.

..... Now that I think even more about this, I think a login page is valid, but SignUp isn't. Admins will register teachers and teachers will register students. Someone shouldn't be allowed to arrive on the site and Sign Up at all.

**Comment by garettarrowood** [comment]
Would you be able to delete the Sign Up endpoint(s) & links and modify the examples you've added to be for login instead?

Let me know if you'd like me to open another issue for that, and/or feel free to put one together yourself. 

Again, thanks for the assistance here!

**Comment by napster235** [comment]
Sure, I can do it. I will close my PR and create a new issue for removing the registration from the app.

**Comment by napster235** [comment]
I think this issue can be closed as well.
